### PR TITLE
Made the DevTools to detect  for mobile mode

### LINKED ISSUE #35: DevTools not detecting then mobile mode enabled
Does not determine the state of devtools when mobile view is enabled.
image

### PR DESCRIPTION
**Pull request by fleventy-5**
Made to detect when dev tools is opened in mobiles mode and table/ipad

Fixes #35 

### For iphone
screenshot1
scrennshot2
### For Android
screenshot3
screenshot4
Browser used: Firefox

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#35: DevTools not detecting then mobile mode enabled](https://issuehunt.io/repos/11125160/issues/35)
---
</details>

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you please explain your changes?

**Comment by sindresorhus** [comment]
You also need to ensure the linting passes.

**Comment by fleventy-5** [comment]
@sindresorhus do you mean I have to edit the code to pass the check.

**Comment by fleventy-5** [comment]
hey @sindresorhus  I have ensured that it passed the linting passes. Anyways if your free can you tell me if there is a tool to automatically  format code that can pass eslinting test. Thanks

**Comment by fleventy-5** [comment]
> Can you please explain your changes?

As for the changes since. When opening in responsive mode(for phones/tablets and even in higher screen sizes) the inner and outer width  or inner height remains the same even wen developer console Is open. So I just wrote a function to check is the user agent is phone and created another function to  execute console.log and get the performance difference before and after it finish to execute console.log and the value will vary depending on whether developers console is open or not. And I used this value to determine whether if the console is open or not.

File: index.js [hunk 1]
```diff
@@ -24,21 +24,61 @@ MIT License
...
+	// Function to check if the given device is a phone or ipad
+	function isPhone() {
+		if (
+			(navigator.userAgent.toLowerCase().indexOf('android') > -1) || (navigator.userAgent.toLowerCase().indexOf('iphone') > -1) ||
+			(navigator.userAgent.toLowerCase().indexOf('ipad') > -1)) {
+			return true;
+		}
+
+		return false;
+	}
```
Comment (papb) [inline index.js:36]:
```suggestion
	function isPhone() {
		return /android|ipad/i.test(navigator.userAgent);
	}
```

File: index.js [hunk 2]
```diff
@@ -24,21 +24,55 @@ MIT License
...
 
+	// Function to check if the given device is a phone or ipad
+	function isPhone() {
+		return /android|ipad/i.test(navigator.userAgent);
+	}
+
+	// Check how long its takes for the given code to execute
+	// It return a value after execution
+	// If value is greater than 60 then dev tool is open
+	// If value is less than 60 then dev tool is not open
```
Comment (sindresorhus) [inline index.js:35]: This sounds very fragile. Is this really reliable enough? The performance difference between a low-end Android phone and the latest iPad is extreme.

File: index.js [hunk 3]
```diff
@@ -24,21 +24,55 @@ MIT License
...
 	const main = ({emitEvents = true} = {}) => {
 		const widthThreshold = window.outerWidth - window.innerWidth > threshold;
 		const heightThreshold = window.outerHeight - window.innerHeight > threshold;
 		const orientation = widthThreshold ? 'vertical' : 'horizontal';
 
-		if (
-			!(heightThreshold && widthThreshold) &&
-			((window.Firebug && window.Firebug.chrome && window.Firebug.chrome.isInitialized) || widthThreshold || heightThreshold)
-		) {
+		if (!(heightThreshold && widthThreshold) && ((window.Firebug && window.Firebug.chrome && window.Firebug.chrome.isInitialized) || widthThreshold || heightThreshold)) {
```
Comment (sindresorhus) [inline index.js:53]: Don't do unrelated changes.

File: index.js [hunk 4]
```diff
@@ -24,21 +24,55 @@ MIT License
...
 			devtools.isOpen = true;
 			devtools.orientation = orientation;
+		} else if (isPhone()) {
+			if (parseInt(checkPerformance(), 0) > 60) {
+				if ((!devtools.isOpen || devtools.orientation !== orientation) && emitEvents) {
+					emitEvent(true, '');
+				}
+
+				devtools.isOpen = true;
+				devtools.orientation = '';
```
Comment (sindresorhus) [inline index.js:67]: ?

File: index.js [hunk 5]
```diff
@@ -24,21 +24,55 @@ MIT License
...
-		) {
+		if (!(heightThreshold && widthThreshold) && ((window.Firebug && window.Firebug.chrome && window.Firebug.chrome.isInitialized) || widthThreshold || heightThreshold)) {
 			if ((!devtools.isOpen || devtools.orientation !== orientation) && emitEvents) {
 				emitEvent(true, orientation);
 			}
 
 			devtools.isOpen = true;
 			devtools.orientation = orientation;
+		} else if (isPhone()) {
+			if (parseInt(checkPerformance(), 0) > 60) {
```
Comment (sindresorhus) [inline index.js:61]: `0` => `10`
